src/main/java/org/spongepowered/api/entity/Item.java

```diff
@@ -25,17 +25,88 @@
 
 package org.spongepowered.api.entity;
 
+import com.google.common.base.Optional;
+import org.spongepowered.api.entity.player.Player;
 import org.spongepowered.api.item.inventory.ItemStack;
 
+import javax.annotation.Nullable;
+
 /**
  * Represents an Item entity.
  */
 public interface Item extends Entity {
 
     /**
-     * Get the {@link ItemStack} that this entity represents.
+     * Get the {@link ItemStack} that this Item represents.
      *
      * @return The represented {@link ItemStack}
      */
     ItemStack getItemStack();
+
+    /**
+     * Gets the number of ticks remaining until this Item can be picked up,
+     * or -1 if this Item has an infinite pickup delay.
+     *
+     * @return The number of ticks remaining, or -1
+     */
+    int getPickupDelay();
+
+    /**
+     * Sets the number of ticks remaining until this Item can be picked up.
+     *
+     * <p>If this Item currently has an infinite pickup delay, the infinite
+     * pickup delay will be removed, and delay will be set instead.</p>
+     *
+     * @param delay The number of ticks remaining
+     */
+    void setPickupDelay(int delay);
+
+    /**
+     * Sets whether this Item has an infinite pickup delay
+     *
+     * @param infinite Whether this Item has an infinite pickup delay
+     */
+    void setInfinitePickupDelay(boolean infinite);
+
+    /**
+     * Gets the number of ticks remaining until this Item despawns,
+     * or -1 if this Item will never despawn.
+     *
+     * @return The number of ticks remaining, or -1
+     */
+    int getDespawnTime();
+
+    /**
+     * Sets the number of ticks remaining until this Item despawns.
+     *
+     * <p>If this Item currently has an infinite despawn delay, the infinite
+     * despawn delay will be removed, and time will be set instead.</p>
+     *
+     * @param time The number of ticks remaining
+     */
+    void setDespawnTime(int time);
+
+    /**
+     * Sets whether this Item never despawns.
+     *
+     * @param infinite Whether this Item never despawns
+     */
+    void setInfiniteDespawnTime(boolean infinite);
+
+    /**
+     * Gets the {@link Player} who threw this Item, if available
+     *
+     * <p>If this Item was not dropped from a player's inventory,
+     * then the thrower will not be available.</p>
+     *
+     * @return The thrower, or Optional.absent()
+     */
+    Optional<Player> getThrower();
```

Comment:
It's completely possible to get the player who dropped the item. Using mixins, code to set the player who threw the item can be added using `@Inject`. Probably on the method `func_146097_a`
Otherwise, a lookup of the player name could be feasible too.